pkg/controllers/management/auth/globalroles/fleetworkspace_role_handler.go

```diff
@@ -0,0 +1,162 @@
+	cr, err := h.crCache.Get(crName)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+
+	if apierrors.IsNotFound(err) {
+		_, err := h.crClient.Create(&v1.ClusterRole{
+			TypeMeta: metav1.TypeMeta{},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: crName,
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion: v3.GlobalRoleGroupVersionKind.GroupVersion().String(),
+						Kind:       v3.GlobalRoleGroupVersionKind.Kind,
+						Name:       globalRole.Name,
+						UID:        globalRole.UID,
+					},
+				},
+				Labels: map[string]string{
+					grOwnerLabel:                wrangler.SafeConcatName(globalRole.Name),
+					controllers.K8sManagedByKey: controllers.ManagerValue,
+				},
+			},
+			Rules: globalRole.InheritedFleetWorkspacePermissions.ResourceRules,
+		})
+
+		return err
+	}
+
+	if !equality.Semantic.DeepEqual(cr.Rules, globalRole.InheritedFleetWorkspacePermissions.ResourceRules) {
+		// undo modifications if cr has changed
+		cr.Rules = globalRole.InheritedFleetWorkspacePermissions.ResourceRules
+		_, err := h.crClient.Update(cr)
+
+		return err
+	}
+
+	return err
+}
+
+func (h *fleetWorkspaceRoleHandler) reconcileWorkspaceVerbs(globalRole *v3.GlobalRole) error {
+	if globalRole.InheritedFleetWorkspacePermissions.WorkspaceVerbs == nil || len(globalRole.InheritedFleetWorkspacePermissions.WorkspaceVerbs) == 0 {
+		return nil
+	}
+	crName := wrangler.SafeConcatName(globalRole.Name, fleetWorkspaceVerbsName)
+	fleetWorkspaces, err := h.fwCache.List(labels.Everything())
+	if err != nil {
+		return fmt.Errorf("unable to list fleetWorkspaces when reconciling globalRole %s: %w", globalRole.Name, err)
+	}
+
+	var workspacesNames []string
+	for _, fleetWorkspace := range fleetWorkspaces {
+		if fleetWorkspace.Name != localFleetWorkspace {
+			workspacesNames = append(workspacesNames, fleetWorkspace.Name)
+		}
+	}
+	rules := []v1.PolicyRule{
+		{
+			Verbs:         globalRole.InheritedFleetWorkspacePermissions.WorkspaceVerbs,
+			APIGroups:     []string{"management.cattle.io"},
+			Resources:     []string{"fleetworkspaces"},
+			ResourceNames: workspacesNames,
+		},
+	}
+	cr, err := h.crCache.Get(crName)
+	if err != nil && !apierrors.IsNotFound(err) {
+		return err
+	}
+	if apierrors.IsNotFound(err) {
+		_, err := h.crClient.Create(&v1.ClusterRole{
+			TypeMeta: metav1.TypeMeta{},
+			ObjectMeta: metav1.ObjectMeta{
+				Name: crName,
+				OwnerReferences: []metav1.OwnerReference{
+					{
+						APIVersion: v3.GlobalRoleGroupVersionKind.GroupVersion().String(),
+						Kind:       v3.GlobalRoleGroupVersionKind.Kind,
+						Name:       globalRole.Name,
+						UID:        globalRole.UID,
+					},
+				},
+				Labels: map[string]string{
+					grOwnerLabel:                wrangler.SafeConcatName(globalRole.Name),
+					controllers.K8sManagedByKey: controllers.ManagerValue,
+				},
+			},
+			Rules: rules,
+		})
+
+		return err
+	} else if !equality.Semantic.DeepEqual(cr.Rules, rules) {
```

Comment:
Nit: It's possible that this isn't as efficient as it could be. DeepEqual on a slice of struct is going to check every struct and look for equality at several levels (including group/resources/verbs). We know that this is going to be a bit more static (with the same resources/groups each time), so we could write something that takes advantage of this to save a few cpu cycles.

I've marked this as a nit (you can ignore if you don't think it's helpful) since:
- This will be a bit more error prone and you'll need to have tests to cover the various cases
- I'd expect that this won't be much faster in most cases.

Reply:
There is nothing preventing the user to manually change the resources/groups, so I think we should also check them. Anyway I would prefer using `equality.Semantic.DeepEqual` since it's already been tested.

Reply:
Just to be clear, I wasn't saying that we shouldn't check the resources/groups, more that we could use a targeted check to look at them more efficiently (i.e. check if they are len(1) and that the first entry is the exact item that we expect).

That being said, I can agree with the idea that we should use a known, well-tested method over something custom, so I'll resolve this and we can keep using what we have.